src/main/java/org/mskcc/cmo/metadb/service/impl/UpdateMessageHandlingServiceImpl.java

```diff
@@ -0,0 +1,287 @@
+    private ValidRequestChecker validRequestChecker;
+
+    private static boolean initialized = false;
+    private static Gateway messagingGateway;
+    private static final Log LOG = LogFactory.getLog(MessageHandlingServiceImpl.class);
+    private final ObjectMapper mapper = new ObjectMapper();
+    private static final ExecutorService exec = Executors.newCachedThreadPool();
+    private static volatile boolean shutdownInitiated;
+
+    private static CountDownLatch requestUpdateFilterHandlerShutdownLatch;
+    private static final BlockingQueue<String> requestUpdateFilterQueue =
+            new LinkedBlockingQueue<String>();
+
+    private static CountDownLatch sampleUpdateFilterHandlerShutdownLatch;
+    private static final BlockingQueue<String> sampleUpdateFilterQueue =
+            new LinkedBlockingQueue<String>();
+
+    @Override
+    public void initialize(Gateway gateway) throws Exception {
+        if (!initialized) {
+            messagingGateway = gateway;
+            setupRequestUpdateFilterHandler(messagingGateway, this);
+            setupSampleUpdateFilterHandler(messagingGateway, this);
+            initializeMessageFilterHandlers();
+            initialized = true;
+        } else {
+            LOG.error("Messaging Handler Service has already been initialized,"
+                    + "ignoring request or sample updates.\n");
+        }
+
+    }
+
+    private void initializeMessageFilterHandlers() throws Exception {
+        requestUpdateFilterHandlerShutdownLatch = new CountDownLatch(NUM_NEW_REQUEST_HANDLERS);
+        final Phaser requestFilterPhaser = new Phaser();
+        requestFilterPhaser.register();
+        for (int lc = 0; lc < NUM_NEW_REQUEST_HANDLERS; lc++) {
+            requestFilterPhaser.register();
+            exec.execute(new RequestUpdateFilterHandler(requestFilterPhaser));
+        }
+        requestFilterPhaser.arriveAndAwaitAdvance();
+
+        sampleUpdateFilterHandlerShutdownLatch = new CountDownLatch(NUM_NEW_REQUEST_HANDLERS);
+        final Phaser sampleFilterPhaser = new Phaser();
+        sampleFilterPhaser.register();
+        for (int lc = 0; lc < NUM_NEW_REQUEST_HANDLERS; lc++) {
+            sampleFilterPhaser.register();
+            exec.execute(new SampleUpdateFilterHandler(sampleFilterPhaser));
+        }
+        sampleFilterPhaser.arriveAndAwaitAdvance();
+    }
+
+    private class RequestUpdateFilterHandler implements Runnable {
+        final Phaser phaser;
+        boolean interrupted = false;
+
+        RequestUpdateFilterHandler(Phaser phaser) {
+            this.phaser = phaser;
+        }
+
+        @Override
+        public void run() {
+            phaser.arrive();
+            while (true) {
+                try {
+                    String requestJson = requestUpdateFilterQueue.poll(100, TimeUnit.MILLISECONDS);
+                    if (requestJson != null) {
+                        Boolean passCheck = validRequestChecker.isValidRequestMetadataJson(requestJson);
+
```

Comment:
extra line

Reply:
hard to tell on this view but if the extra line hasn't been removed already then go ahead and remove it, otherwise feel free to hit the "resolve conversation" button here